pkg/controller/controller_test.go

```diff
@@ -497,83 +509,105 @@ func (s *ControllerSuite) TestSynchronousFailure(c *C) {
 	c.Assert(err, IsNil)
 }
 
+func (s *ControllerSuite) TestNilPrometheusRegistry(c *C) {
+	config, _ := kube.LoadConfig()
+	ctrl := New(config, nil)
+	c.Assert(ctrl, NotNil)
+	c.Assert(ctrl.metrics, IsNil)
+}
+
 func (s *ControllerSuite) TestExecActionSet(c *C) {
 	for _, pok := range []string{"StatefulSet", "Deployment"} {
 		for _, tc := range []struct {
-			funcNames []string
-			args      [][]string
-			name      string
-			version   string
+			funcNames        []string
+			args             [][]string
+			name             string
+			version          string
+			metricResolution string
 		}{
 			{
-				funcNames: []string{testutil.WaitFuncName},
-				name:      "WaitFunc",
-				version:   kanister.DefaultVersion,
+				funcNames:        []string{testutil.WaitFuncName},
+				name:             "WaitFunc",
+				version:          kanister.DefaultVersion,
+				metricResolution: "success",
```

Comment:
We could use the same label constants here

Reply:
Thank you for catching this. Updated.